src/main/java/org/opensearch/sdk/rest/ExtensionRestPathRegistry.java

```diff
@@ -133,6 +133,24 @@ private void registerWithDeprecatedHandler(
         registerAsDeprecatedHandler(deprecatedMethod, deprecatedPath, handler, deprecationMessage);
     }
 
+    /**
+     * Register a REST handler to handle a method and route in this extension's path registry.
+     *
+     * @param method  The method to register.
+     * @param path  The path to register. May include named wildcards.
+     * @param name  An optional name of the REST handler
+     * @param extensionRestHandler  The RestHandler to handle this route
+     */
+    public void registerHandler(Method method, String path, String name, ExtensionRestHandler extensionRestHandler) {
+        pathTrie.insertOrUpdate(
+            path,
+            new SDKMethodHandlers(path, extensionRestHandler, method),
+            (mHandlers, newMHandler) -> mHandlers.addMethods(extensionRestHandler, method)
+        );
+        String restPathWithName = restPathToString(method, path, name);
+        registeredPaths.add(restPathWithName);
```

Comment:
We are missing adding paths to deprecated paths here. One way to handle this would be to merge both the `registerHandler` methods and add a check for `name`.